core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2105,7 +2103,7 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
             s1.putAttribute("workspaceId", OpenCensusAttributeValue.stringAttributeValue(workspaceId))
 
             traceDBIOWithParent("getPolicySyncStatus", s1)(_ => DBIO.from(samDAO.getPolicySyncStatus(SamResourceTypeNames.billingProject, workspaceRequest.namespace, SamBillingProjectPolicyNames.owner, userInfo).map(_.email))).flatMap { projectOwnerPolicyEmail =>
-              traceDBIOWithParent("setupGoogleProject", s1)(_ => DBIO.from(setupGoogleProject(billingProject, s1))).flatMap { case (googleProjectId, googleProjectNumber) =>
+              traceDBIOWithParent("setupGoogleProject", s1)(_ => DBIO.from(setupGoogleProject(billingProject, s1, workspaceId))).flatMap { case (googleProjectId, googleProjectNumber) =>
```

Comment:
Should we consider using `WorkspaceName` as the `handoutRequestId` here?  The `workspaceId` is a UUID we just generated here and there isn't much to tie it back to the original request.  For example, if the `saveNewWorkspace` action should fail on the next line, we won't know the `workspaceId` except for maybe/hopefully in the logs it will be in the failed SQL statement or something.  If we use `workspaceRequest.toWorkspaceName` here, it will hopefully make it easier to track/correlate the request.  

Oh, and I guess all of this is on the assumption that the purpose of the `handoutRequestId` is purely for tracking/auditing/troubleshooting requests for Resources from RBS

Reply:
`handoutRequestId` is not at all for tracking/auditing/troubleshooting, it is functional. RBS will always give back the same google project for a given `handoutRequestId`. It is a way to avoid dropping resources on the floor in the case a retries.

So the code is right as is. Consider the case that a user creates a workspace, deletes it, then creates it again. We want to be sure RBS gives us back a different project.